cmd/infpool_miner_max_borrow.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright © 2023 Glif LTD
+*/
+package cmd
+
+import (
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/briandowns/spinner"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/filecoin-project/go-address"
+	"github.com/glifio/go-pools/mstat"
+	psdk "github.com/glifio/go-pools/sdk"
+	denoms "github.com/glifio/go-pools/util"
+	"github.com/glifio/go-pools/vc"
+	"github.com/spf13/cobra"
+)
+
+var infpoolMinerQuote = &cobra.Command{
+	Use:   "miner-quote <miner-addr>",
+	Short: "Returns the amount of FIL the miner can",
+	Args:  cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		minerAddr, err := address.NewFromString(args[0])
+		if err != nil {
+			logFatal(err)
+		}
+
+		fmt.Printf("Generating a quote for %s\n", minerAddr)
+
+		s := spinner.New(spinner.CharSets[9], 100*time.Millisecond)
+		s.Start()
+		defer s.Stop()
+
+		lCli, closer, err := PoolsSDK.Extern().ConnectLotusClient()
+		if err != nil {
+			logFatal(err)
+		}
+		defer closer()
+
+		tipset, err := lCli.ChainHead(cmd.Context())
+		if err != nil {
+			logFatal(err)
+		}
+
+		minerstat, err := mstat.ComputeMinerStats(cmd.Context(), minerAddr, tipset, lCli)
+		if err != nil {
+			logFatal(err)
+		}
+
+		edr := new(big.Int).Add(minerstat.ExpectedDailyReward, new(big.Int).Div(minerstat.VestingFunds, big.NewInt(180)))
+
+		agentData := &vc.AgentData{
```

Comment:
How can some of these values be hard coded and the SP get a meaningful quote?

Reply:
They're not all hardcoded - just the ones that would apply to an agent (like current amount of principal)